torch/csrc/autograd/python_engine.cpp

```diff
@@ -81,7 +81,7 @@ void PythonEngine::thread_init(int device, const std::shared_ptr<ReadyQueue>& re
 
 #ifdef IS_PYTHON_3_9_PLUS
   // Do not call PyEval_RestoreThread, PyThreadState_[Clear|DeleteCurrent] if runtime is finalizing
-  if (_Py_IsFinalizing()) {
+  if (!Py_IsInitialized()) {
```

Comment:
Sort of same as the above
```suggestion
  if (!Py_IsInitialized() || _Py_IsFinalizing()) {
```